perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_attribute_def_def_freeipaGroupName.java

```diff
@@ -0,0 +1,73 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.Group;
+import cz.metacentrum.perun.core.api.Vo;
+import cz.metacentrum.perun.core.api.exceptions.ConsistencyErrorException;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.VoNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.GroupAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.GroupAttributesModuleImplApi;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Module for attribute freeipaGroupName
+ *
+ * @author Peter Balcirak <peter.balcirak@gmail.com>
+ * @date 8.5.2016
+ */
+public class urn_perun_group_attribute_def_def_freeipaGroupName  extends GroupAttributesModuleAbstract implements GroupAttributesModuleImplApi  {
+
+	private static final Pattern pattern = Pattern.compile("^[a-zA-Z0-9_.][a-zA-Z0-9_.-]{0,252}[a-zA-Z0-9_.$-]?$");
+
+	@Override
+	public void checkAttributeValue(PerunSessionImpl sess, Group group, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+
+		//prepare group name and check its format
+		String groupName = (String) attribute.getValue();
+		if (groupName == null) {
+			throw new WrongAttributeValueException(attribute, group, "Attribute can't be null.");
+		}
+
+		Matcher match = pattern.matcher(groupName);
+
+		if (!match.matches()) {
+			throw new WrongAttributeValueException(attribute, group, "Bad format of attribute freeipaGroupName. It has to match pattern ^[a-zA-Z0-9_.][a-zA-Z0-9_.-]{0,252}[a-zA-Z0-9_.$-]?$");
+		}
+
+		//get all groups from vo and check if attribute group name is already used in any of the groups
+		try{
```

Comment:
This is not correct. We need to check uniqueness of groupName on facility.
Facility can have multiple resources. Every resource can be from different VO.
Two Groups in one VO can be assigned to different resources on different facilities.